### ISSUE TITLE
[BUG] PreTrainedModel state set to `eval` rather than `train` when fine-tuning an HF-text models

### ISSUE THREAD

**Issue by SnoopKilla** [autogluon/autogluon#4965]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

Since Lightning v2.2, the the `Trainer` captures the mode of every submodule before switching to validation, and restores each such mode when validation ends (see [here](https://github.com/Lightning-AI/pytorch-lightning/pull/18951) for details).
Since when using `AutoModel.from_pretrained()` the loaded `PreTrainedModel` is in `eval` mode, the backbone of the `HFAutoModelForTextPrediction` **will be in eval mode during the whole training**.
This is not the expected behavior as it will cause, e.g., dropout and batch normalisation layers to not work properly.

**Expected behavior**

The expected behavior is to have the backbone of `HFAutoModelForTextPrediction` in `train` mode during the training steps of the fine-tuning process.

**To Reproduce**
Minimal example: [gist](https://gist.github.com/SnoopKilla/d2eebdbd607aa81270d0c319fb305446)

**Steps to reproduce**:
1. Create a conditional breakpoint at line 230 of `autogluon/multimodal/optimization/lit_module.py` with expression `self.training` (this will make sure the validation sanity checking steps are skipped).
2. Run example code in debug mode.
3. Check from the debug console that `self.training` is `True` (`LitModule` is in training mode), `self.model.training` is `True` (`HFAutoModelForTextPrediction` is in training mode) but `self.model.model.training` is `False` (the loaded `PreTrainedModel` from HuggingFace is in eval mode).

**Screenshots / Logs**

<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/50f3a3e8-2914-4095-8621-0a1e223f8892" />

**Installed Versions**

<details>

```
INSTALLED VERSIONS
------------------
date                   : 2025-03-07
time                   : 19:49:44.492170
python                 : 3.11.9.final.0
OS                     : Linux
OS-release             : 5.10.234-205.895.amzn2int.x86_64
Version                : #1 SMP Tue Feb 11 11:40:58 UTC 2025
machine                : x86_64
processor              : x86_64
num_cores              : 16
cpu_ram_mb             : 31395.00390625
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 55669

accelerate             : 0.34.2
autogluon              : 1.2
autogluon.common       : 1.2
autogluon.core         : 1.2
autogluon.features     : 1.2
autogluon.multimodal   : 1.2
autogluon.tabular      : 1.2
autogluon.timeseries   : 1.2
boto3                  : 1.37.8
catboost               : 1.2.7
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.3
fastai                 : 2.7.18
fugue                  : 0.9.1
gluonts                : 0.16.0
huggingface-hub        : 0.29.2
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.6
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.5.0
lightning              : 2.5.0.post0
matplotlib             : 3.10.1
mlforecast             : 0.13.4
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.8.1
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.10.15
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.1.0
psutil                 : 6.1.1
pyarrow                : 19.0.1
pytesseract            : 0.3.10
pytorch-lightning      : 2.5.0.post0
pytorch-metric-learning: 2.3.0
ray                    : 2.39.0
requests               : 2.32.3
scikit-image           : 0.24.0
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.15.2
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.7.5
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.19.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.5.1
torchmetrics           : 1.2.1
torchvision            : 0.20.1
tqdm                   : 4.67.1
transformers           : 4.49.0
utilsforecast          : 0.2.4
vowpalwabbit           : None
xgboost                : 2.1.4
```

</details>

**Comment by tonyhoo** [comment]
Resolving as this issue is fixed and will be released in version 1.3
